homeassistant/components/media_player/frontier_silicon.py

```diff
@@ -0,0 +1,247 @@
+"""
+Support for Frontier Silicon Devices (Medion, Hama, Auna,...).
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.frontier_silicon/
+"""
+import logging
+
+import requests
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    SUPPORT_NEXT_TRACK, SUPPORT_PAUSE, SUPPORT_PREVIOUS_TRACK, SUPPORT_SEEK,
+    SUPPORT_PLAY_MEDIA, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET,
+    SUPPORT_VOLUME_STEP, SUPPORT_STOP, SUPPORT_TURN_OFF, SUPPORT_TURN_ON,
+    SUPPORT_PLAY, SUPPORT_SELECT_SOURCE, MediaPlayerDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (
+    STATE_OFF, STATE_PLAYING, STATE_PAUSED, STATE_UNKNOWN,
+    CONF_HOST, CONF_PORT, CONF_PASSWORD)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['fsapi==0.0.7']
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_FRONTIER_SILICON = SUPPORT_PAUSE | SUPPORT_VOLUME_SET | \
+    SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_STEP | \
+    SUPPORT_PREVIOUS_TRACK | SUPPORT_NEXT_TRACK | SUPPORT_SEEK | \
+    SUPPORT_PLAY_MEDIA | SUPPORT_PLAY | SUPPORT_STOP | SUPPORT_TURN_ON | \
+    SUPPORT_TURN_OFF | SUPPORT_SELECT_SOURCE
+
+DEFAULT_HOST = '192.168.1.11'
+DEFAULT_PORT = 80
+DEFAULT_PASSWORD = '1234'
+DEVICE_URL = 'http://{0}:{1}/device'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_PASSWORD, default=DEFAULT_PASSWORD): cv.string,
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Frontier Silicon platform."""
+    if discovery_info is not None:
+        add_devices(
+            [FSAPIDevice(discovery_info, DEFAULT_PASSWORD)],
+            update_before_add=True)
+        return True
+
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    password = config.get(CONF_PASSWORD)
+
+    try:
+        if host and port and password:
```

Comment:
No need for this if you use `vol.Required` instead of `vol.Optional` in PLATFORM_SCHEMA.